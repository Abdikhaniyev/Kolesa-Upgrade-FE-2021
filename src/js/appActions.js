export const createApp = () => {
    const appElement = document.createElement('div');

    appElement.setAttribute('id', 'app');
    document.querySelector('body').append(appElement);
};

export const appSetContent = (content) => {
    const appElement = document.querySelector('#app');

    appElement.innerHTML = content;
};

export const showApp = () => {
    const appElement = document.querySelector('#app');

    appElement.style.display = 'block';
};
