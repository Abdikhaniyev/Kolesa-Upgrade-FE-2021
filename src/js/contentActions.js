export const setContentText = (e, text) => {
    e.currentTarget.textContent = text;
};

export const addEventListenerToButtons = (parent, listener) => {
    Array.from(parent.querySelectorAll('button')).forEach((button) => {
        button.addEventListener('click', listener);
    });
};
