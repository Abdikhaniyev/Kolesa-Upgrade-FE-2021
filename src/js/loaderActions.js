export const showLoader = () => {
    document.querySelector('#loader').style.display = 'block';
};

export const hideLoader = () => {
    document.querySelector('#loader').style.display = 'none';
};

export const createLoader = () => {
    const loader = document.createElement('div');

    loader.setAttribute('id', 'loader');
    document.querySelector('body').append(loader);

    return loader;
};
