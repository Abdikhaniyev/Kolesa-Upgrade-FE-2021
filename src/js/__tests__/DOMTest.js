/**
 * @jest-environment jsdom
 */

import { createApp, showApp, appSetContent } from '../appActions';
import { addEventListenerToButtons } from '../contentActions';
import { createLoader, hideLoader, showLoader } from '../loaderActions';

const body = document.querySelector('body');

describe('Группа тестов для DOM.', () => {
    beforeAll(() => {
        createApp();
        showApp();
    });

    afterAll(() => {
        appSetContent('');
    });

    test('Тест на добавление контента', () => {
        appSetContent('Kolesa Group');

        expect(body).toMatchSnapshot();
    });

    test('Тест на отображение и скрытие лоадера', () => {
        const loader = createLoader();

        showLoader();
        expect(loader).toBeDefined();
        expect(loader.style.display).toBe('block');

        hideLoader();
        expect(loader.style.display).toBe('none');
    });

    test('Тест на добавление событий', () => {
        const clickMock = jest.fn();
        const container = document.createElement('div');

        for (let i = 0; i < 3; i += 1) {
            const button = document.createElement('button');

            container.append(button);
        }

        body.append(container);

        const buttons = body.querySelectorAll('button');

        addEventListenerToButtons(body, clickMock);
        buttons.forEach((button) => {
            button.click();
            expect(clickMock).toHaveBeenCalled();
        });

        expect(clickMock).toHaveBeenCalledTimes(3);
    });
});
