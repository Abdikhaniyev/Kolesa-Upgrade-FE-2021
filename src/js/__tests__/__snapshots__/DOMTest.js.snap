// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Группа тестов для DOM. Тест на добавление контента 1`] = `
<body>
  <div
    id="app"
    style="display: block;"
  >
    Kolesa Group
  </div>
</body>
`;
